airflow/providers/cncf/kubernetes/operators/spark_kubernetes_operator.py

```diff
@@ -0,0 +1,83 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Optional
+
+import yaml
+from kubernetes import client
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.cncf.kubernetes.hooks.kubernetes_hook import Kuberneteshook
+from airflow.utils.decorators import apply_defaults
+
+
+class SparkKubernetesOperator(BaseOperator):
+    """
+    Creates sparkApplication object in kubernetes cluster:
+       .. seealso::
+        For more detail about Spark Application Object have a look at the reference:
+        https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/docs/api-docs.md#sparkapplication
+
+    :param sparkapplication_file: filepath to kubernetes custom_resource_definition of sparkApplication
+    :type sparkapplication_file:  str
+    :param namespace: kubernetes namespace to put sparkApplication
+    :type namespace: str
+    :param conn_id: the connection to Kubernetes cluster
+    :type conn_id: str
+    """
+
+    template_fields = ['sparkapplication_file', 'namespace']
+    template_ext = ('yaml', 'yml', 'json')
+    ui_color = '#f4a460'
+
+    @apply_defaults
+    def __init__(self,
+                 sparkapplication_file: str,
+                 namespace: Optional[str] = None,
+                 conn_id: str = 'kubernetes_default',
+                 *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.sparkapplication_file = sparkapplication_file
+        self.namespace = namespace
+        self.conn_id = conn_id
+
+    def execute(self, context):
+        self.log.info("Creating sparkApplication")
+        hook = Kuberneteshook(conn_id=self.conn_id)
+        api_client = hook.get_conn()
+        api = client.CustomObjectsApi(api_client)
```

Comment:
This feels like it should be in the hook, not the operator.

Reply:
but then I'll need to write code in the hook for every Kubernetes API instead of just return general API connection and the operator decide which API kind he opens.
what do you think is right?

Reply:
Oh sorry I was misreading this code - `client` was confusing me as a module name